### ISSUE TITLE
Compilation error in openSUSE (version 0.9.4, openSUSE Factory / Tumbleweed)

### ISSUE THREAD

**Issue by LazyKent** [lbellonda/qxmledit#37]
I get compilation error in openSUSE Factory / Tumbleweed.

```
 g++ -c -include ../build/lib/obj/QXmlEditWidget-0.9.4 -pipe -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -g -Wall -Werror -std=c++11 -O2 -Wall -W -D_REENTRANT -fPIC -DLIBQXMLEDIT_LIBRARY -DUNIX_RESOURCES -DUNIX_RESOURCE_PATH=/usr/share/qxmledit/ -DUNIX_DOC_PATH=/usr/share/doc/packages/qxmledit/ -DQT_NO_DEBUG -DQT_SVG_LIB -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_XML_LIB -DQT_XMLPATTERNS_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I. -Iglobals/includes -I./globals/framework -Iglobals/includes -Iframework/include -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtSvg -isystem /usr/include/qt5/QtPrintSupport -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtXml -isystem /usr/include/qt5/QtXmlPatterns -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtCore -I../build/lib/moc -I../build/lib/ui -I/usr/lib64/qt5/mkspecs/linux-g++ -o ../build/lib/obj/edittextnode.o edittextnode.cpp
 elementitemdelegate.cpp: In member function 'virtual void ElementItemDelegate::paint(QPainter*, const QStyleOptionViewItem&, const QModelIndex&) const':
 elementitemdelegate.cpp:51:46: error: 'QStyleOptionViewItemV4' is deprecated [-Werror=deprecated-declarations]
          QStyleOptionViewItemV4 option4(option);
                                               ^
 In file included from /usr/include/qt5/QtWidgets/qabstractitemdelegate.h:44:0,
                  from /usr/include/qt5/QtWidgets/qabstractitemview.h:46,
                  from /usr/include/qt5/QtWidgets/qheaderview.h:43,
                  from /usr/include/qt5/QtWidgets/QHeaderView:1,
                  from globals/includes/qtincludes.h:37,
                  from globals/includes/xmlEdit.h:55,
                  from precompiled_lib.h:27:
 /usr/include/qt5/QtWidgets/qstyleoption.h:441:48: note: declared here
  typedef Q_DECL_DEPRECATED QStyleOptionViewItem QStyleOptionViewItemV4;
                                                 ^~~~~~~~~~~~~~~~~~~~~~
```

The full build log: https://build.opensuse.org/build/openSUSE:Factory/standard/x86_64/qxmledit/_log

Thanks.

**Comment by lbellonda** [comment]
Hello, what version of Qt are you using, 5.7? What version of gcc?
As a workaround you can comment the options to generate errors from warnings in the project:
QMAKE_CXXFLAGS += -Wall
QMAKE_CXXFLAGS +=-Werror
until the problem is solved.

**Comment by LazyKent** [comment]
Hello. Found in the log:
gcc-c++-6-2.2
Qt5-5.7.0

**Comment by lbellonda** [comment]
QStyleOptionViewItemV4 has been declared deprecated in Qt 5.7. The fix should be simple, however compatibility with Qt4.7 deserves some test before the commit.

**Comment by lbellonda** [comment]
Committed a fix on the branch "maint_0.9.4".
Waiting for a feedback to include the fixes in a new release.

**Comment by LazyKent** [comment]
![qxmledit-0 9 5-snapshot](https://cloud.githubusercontent.com/assets/1275149/19539780/63bee77e-966c-11e6-85ef-530437debb48.png)
I have tested the "maint_0.9.4" branch.
Compiled successfully.

Also tested  a compilation in openSUSE 13.2 (using Qt4 v4.8.6).

Everything is OK.

**Comment by lbellonda** [comment]
The fix will be incorporated in the upcoming 0.9.5 release.

**Comment by lbellonda** [comment]
Fixed in 0.9.5
